tensorly/utils/jointdiag.py

```diff
@@ -0,0 +1,182 @@
+from itertools import combinations
+import tensorly as tl
+
+# Authors: Aaron Meyer <a@ameyer.me>
+#          Sebastian Seljak
+
+# License: BSD 3 clause
+
+
+def jointdiag(
+    X,
+    max_iter: int = 50,
+    threshold: float = 1e-10,
+    verbose: bool = False,
+):
+    """
+    Jointly diagonalizes n matrices, organized in tensor of dimension (k,k,n).
+    Returns the diagonalized matrices, along with the transformation matrix.
+
+    T. Fu and X. Gao, “Simultaneous diagonalization with similarity transformation for
+    non-defective matrices”, in Proc. IEEE International Conference on Acoustics, Speech
+    and Signal Processing (ICASSP 2006), vol. IV, pp. 1137-1140, Toulouse, France, May 2006.
```

Comment:
Can we make it as a proper reference [1]_ and add it a reference section of the doctoring (see e.g. https://github.com/tensorly/tensorly/blob/d4652c8da8d6dfe8484ae282bccddf9db053d90c/tensorly/decomposition/_cp.py#L681 )